helios-client/src/main/java/com/spotify/helios/common/descriptors/ServicePortParameters.java

```diff
@@ -21,27 +21,56 @@
 
 package com.spotify.helios.common.descriptors;
 
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Objects;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
 
 /**
  * A class that used to do something, but now is just effectively a placeholder in a Map
  * that is treated as a Set.  It is represented by the empty map in JSON.
  */
-@JsonSerialize
+@JsonInclude(JsonInclude.Include.NON_NULL)
 public class ServicePortParameters extends Descriptor {
+  private final List<String> tags;
+
+  public ServicePortParameters(@JsonProperty("tags") @Nullable final List<String> tags) {
```

Comment:
Should tags be a List rather than a Set as the description [here](https://github.com/spotify/helios/pull/370/files#diff-366122efb90066c16dfb5eb26a1ff264R33) says?